base/collection_view.go

```diff
@@ -379,3 +381,62 @@ func asViewConsistency(value interface{}) gocb.ViewScanConsistency {
 	}
 
 }
+
+// GetGoCBAgent returns the underlying agent from gocbcore
+func (c *Collection) getGoCBAgent() (*gocbcore.Agent, error) {
+	return c.Bucket().Internal().IORouter()
+
+}
+
+// getCollectionID returns the gocbcore CollectionID for the current collection
+func (c *Collection) getCollectionID() (uint32, error) {
```

Comment:
Is it correct that collectionID is immutable on server?  If so it might make sense to store the collectionID on Collection after we retrieve it, in case we have future use cases where we end up calling getCollection more than once for a Collection instance.

Reply:
I believe this is immutable, but @bbrks should confirm.

Reply:
It is immutable, especially for a connection, so cached the value.